### ISSUE TITLE
How do we take care 401 Unauthorized

### ISSUE THREAD

**Issue by tianxu0836** [sahat/satellizer#389]
Is it possible to take care 401 Unauthorized when token expires?

**Comment by sahat** [comment]
Yes, but you will have to think about the implementation since it can vary from application to application. This will all have to be done on the back-end where you check the validity of a token. I could perhaps add it to Satellizer back-end examples if you can provide a reasonable generalized implementation for what to do when token is expired.

As for the client-side, Satellizer doesn't handle token expiration logic either other than checking if it has expired or not when you call `isAuthenticated()`. :

```js
shared.isAuthenticated = function() {
          var token = storage.get(tokenName);

          if (token) {
            if (token.split('.').length === 3) {
              var base64Url = token.split('.')[1];
              var base64 = base64Url.replace('-', '+').replace('_', '/');
              var exp = JSON.parse($window.atob(base64)).exp;
              if (exp) {
                return Math.round(new Date().getTime() / 1000) <= exp;
              }
            }
            return true;
          }
          return false;
        };

```

In other words it doesn't care if you were never logged in or if a token has expired, it will return `false` in both cases.
